Support for add relu functional module

raghuramank100 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26612 Support for add relu functional module**

Add support for add relu functional module, this allows for fusion of add and relu quantized operations

Differential Revision: [D17518268](https://our.internmc.facebook.com/intern/diff/D17518268/)

raghuramank10000 committed:
Support for add relu functional module

Add support for add relu functional module, this allows for fusion of add and relu quantized operations

Differential Revision: [D17518268](https://our.internmc.facebook.com/intern/diff/D17518268/)

[ghstack-poisoned]

jerryzh168 commented on test/common_quantization.py:
```diff
@@ -451,22 +451,27 @@ def __init__(self, quantized=False):
 
     @classmethod
     def from_float(cls, mod):
         new_mod = cls(quantized=True)
```
why do you need to do this? isn't this done by the flow?

raghuramank100 replied:
Rewrote the tests to match how we test other modules

jerryzh168 reviewed (changes requested):
I think there is some problem with original test

dskhudia reviewed (approved):
LGTM!

raghuramank10000 committed:
Update on "Support for add relu functional module"


Add support for add relu functional module, this allows for fusion of add and relu quantized operations

Differential Revision: [D17518268](https://our.internmc.facebook.com/intern/diff/D17518268/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Support for add relu functional module"


Add support for add relu functional module, this allows for fusion of add and relu quantized operations

Differential Revision: [D17518268](https://our.internmc.facebook.com/intern/diff/D17518268/)

[ghstack-poisoned]

raghuramank10000 committed:
Update on "Support for add relu functional module"


Add support for add relu functional module, this allows for fusion of add and relu quantized operations

Differential Revision: [D17518268](https://our.internmc.facebook.com/intern/diff/D17518268/)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
Looks good

zafartahirov commented on test/test_quantization.py:
```diff
@@ -670,31 +670,35 @@ def test_single_layer(self):
+    def test_functional_module(self, train_mode):
+        model = ModelWithFunctionals()
+        xq = torch.quantize_per_tensor(torch.rand(10, 1, dtype=torch.float), 0.01, 30, torch.quint8)
+        self.checkScriptable(model, [(xq.dequantize(), xq.dequantize())], check_save_load=True)
```
nit: factor out the `xq.dequantize()` -- it is used in multiple places.

z-a-f reviewed (approved):
LGTM
